test/testclientsideencryption.cpp

```diff
@@ -132,6 +137,99 @@ private slots:
         // THEN
         QCOMPARE(data, originalData);
     }
+
+    void testStreamingDecryptor()
+    {
+        QTemporaryFile dummyInputFile;
+
+        QVERIFY(dummyInputFile.open());
+
+        const auto dummyFileRandomContents = EncryptionHelper::generateRandom(272);
+
+        QCOMPARE(dummyInputFile.write(dummyFileRandomContents), dummyFileRandomContents.size());
+
+        const auto generateHash = [](const QByteArray &data) {
+            QCryptographicHash hash(QCryptographicHash::Sha1);
+            hash.addData(data);
+            return hash.result();
+        };
+
+        const QByteArray originalFileHash = generateHash(dummyFileRandomContents);
+
+        QVERIFY(!originalFileHash.isEmpty());
+
+        dummyInputFile.close();
+        QVERIFY(!dummyInputFile.isOpen());
+
+        const auto encryptionKey = EncryptionHelper::generateRandom(16);
+        const auto initializationVector = EncryptionHelper::generateRandom(16);
+
+        // test normal file encryption/decryption
+        QTemporaryFile dummyEncryptionOutputFile;
+
+        QByteArray tag;
+
+        QVERIFY(EncryptionHelper::fileEncryption(encryptionKey, initializationVector, &dummyInputFile, &dummyEncryptionOutputFile, tag));
+        dummyInputFile.close();
+        QVERIFY(!dummyInputFile.isOpen());
+
+        dummyEncryptionOutputFile.close();
+        QVERIFY(!dummyEncryptionOutputFile.isOpen());
+
+        QTemporaryFile dummyDecryptionOutputFile;
+
+        QVERIFY(EncryptionHelper::fileDecryption(encryptionKey, initializationVector, &dummyEncryptionOutputFile, &dummyDecryptionOutputFile));
+        QVERIFY(dummyDecryptionOutputFile.open());
+        const auto dummyDecryptionOutputFileHash = generateHash(dummyDecryptionOutputFile.readAll());
+        QCOMPARE(dummyDecryptionOutputFileHash, originalFileHash);
+
+        // test streaming decryptor
+        EncryptionHelper::StreamingDecryptor streamingDecryptor(encryptionKey, initializationVector, dummyEncryptionOutputFile.size());
+        QVERIFY(streamingDecryptor.isInitialized());
+
+        QBuffer chunkedOutputDecrypted;
+        QVERIFY(chunkedOutputDecrypted.open(QBuffer::WriteOnly));
+
+        QVERIFY(dummyEncryptionOutputFile.open());
+
+        QByteArray pendingBytes;
+
+        const int randBytesMin = QRandomGenerator::global()->bounded(1, 10);
```

Comment:
not sure how this random generator works exactly, but is ensured that this test is deterministic or will there be some randomness?

Reply:
> not sure how this random generator works exactly, but is ensured that this test is deterministic or will there be some randomness?

@FlexW It's going to imitate random min size of bytes to read. Emulating the number of bytes that may arrive. Different number of bytes may arrive, depending on network condition, etc.

Reply:
Okay. I think we should not have randomness in our tests. Otherwise, it might sometimes fail and sometimes not, and if it fails it will be difficult to diagnose.

Reply:
@FlexW Well, I would argue. It depends on what kind of randomness. In this case, randomness makes a test more reliable. Also, it makes the code that we are testing more reliable. E.G. when I have introduced this randomness, I have noticed tests failing 1 out of 3. This was due to the code I am testing was not expecting randomness and would fail should this randomness happen during real-life execution. So, I had to fix my code. I'd want to keep randomness in this test, otherwise, the test would be useless, as, it will always be succeeding like it was doing before my recent improvements to the tested code.

Reply:
@FlexW Ok, I guess I can just supply 3-4 constant numbers and perform test with multiple values but always same